examples/qwen_3_vl/qwen_3_vl.zig

```diff
@@ -0,0 +1,1211 @@
+        const w_idxs = zml.Tensor.linspace(.{ .start = 0, .end = num_grid_per_side - 1, .steps = w }, .f32);
+        const h_floor = h_idxs.floor();
+        const w_floor = w_idxs.floor();
+
+        // Build the min and max clamp for the height and width
+        const min_clamp_h = zml.Tensor.constant(.{h}, zml.Data.init(.f32, 0));
+        const max_clamp_h = zml.Tensor.constant(.{h}, zml.Data.init(.f32, num_grid_per_side - 1));
+        const min_clamp_w = zml.Tensor.constant(.{w}, zml.Data.init(.f32, 0));
+        const max_clamp_w = zml.Tensor.constant(.{w}, zml.Data.init(.f32, num_grid_per_side - 1));
+
+        // Build the ceil and floor for the height and width
+        const h_ceil = h_floor.add(tensor_filled_1_h).clamp(min_clamp_h, max_clamp_h);
+        const w_ceil = w_floor.add(tensor_filled_1_w).clamp(min_clamp_w, max_clamp_w);
+
+        // Build the difference between the indices and the floor for the height and width -> delta with the grid points
+        const dh = h_idxs.sub(h_floor);
+        const dw = w_idxs.sub(w_floor);
+
+        // Build the meshgrid for the height and width
+        const tensor_filled_1_h_v = zml.Tensor.constant(.{ h, w }, zml.Data.init(.f32, 1));
+        const d_tensors_meshgrid = [2]Tensor{ dh, dw };
+        const floor_tensors_meshgrid = [2]Tensor{ h_floor, w_floor };
+        const ceil_tensors_meshgrid = [2]Tensor{ h_ceil, w_ceil };
+        const dhw_grid = zml.torch.meshgrid(2, d_tensors_meshgrid, .ij);
+        const floorhw_grid = zml.torch.meshgrid(2, floor_tensors_meshgrid, .ij);
+        const ceilhw_grid = zml.torch.meshgrid(2, ceil_tensors_meshgrid, .ij);
+
+        // Compute the weights for the height and width
+        const w11 = dhw_grid[0].mul(dhw_grid[1]);
+        const w10 = dhw_grid[0].sub(w11);
+        const w01 = dhw_grid[1].sub(w11);
+        const w00 = tensor_filled_1_h_v.sub(dhw_grid[0]).sub(w01);
+        const h_list = [4]Tensor{ floorhw_grid[0], floorhw_grid[0], ceilhw_grid[0], ceilhw_grid[0] };
+        const w_list = [4]Tensor{ floorhw_grid[1], ceilhw_grid[1], floorhw_grid[1], ceilhw_grid[1] };
+
+        // Stack the height and width lists
+        const h_grid = zml.Tensor.stack(&h_list, 0, .layers);
+        const w_grid = zml.Tensor.stack(&w_list, 0, .layers);
+
+        const h_grid_idx = h_grid.scale(num_grid_per_side);
+        const indices = h_grid_idx.add(w_grid).reshape(.{ 4, -1 }).convert(.i32);
+        var weights = zml.Tensor.stack(&[4]Tensor{ w00, w01, w10, w11 }, 0, .layers).reshape(.{ 4, -1, 1 });
+        const embeds = zml.call(self.pos_embed, .forward, .{indices});
+        const weights_embed = embeds.convert(.f32).mul(weights.repeat1d(-1, embedding_dim));
+        const combined = weights_embed.sum(0);
+        const combined_reshape = combined.reshape(.{ @divExact(h, m_size), m_size, @divExact(w, m_size), m_size, embedding_dim });
```

Comment:
you should use `splitAxis` instead of reshape, it is less error prone:
 `combined.splitAxis(.h, .{ .h = .auto, .h_patch = m_size }).splitAxis(.w,  .{ .w = .auto, .w_patch = m_size })`